### PR TITLE
refactor(connlib): restructure directory for consistency

### PR DESCRIPTION
**Pull request by thomaseizinger**
We flatten the directory hierarchy within `rust/connlib` and rename some crates for consistency. In particular:

- Everything that is directly related to the _embedded_ library is called `connlib-XYZ`, e.g. `connlib-client-android`.
- The gateway is now called `firezone-gateway` and lives outside of `connlib` because it uses connlib but is a firezone product.
- The headless client is now called `firezone-headless-client` and live outside of `connlib` too. It is not an official product (I think?) but it is of the same nature as the gateway.
- If left the `firezone-tunnel` as-is despite it being within `connlib/` because I thought it had a better ring to it and in a way, the actual tunnel _is_ the product.

Ideally, `firezone-tunnel` would not depend on anything within `connlib` (it currently depends on `connlib-shared`) but that is only possible once we move away from callbacks in favor of events and state machines.

Patch-by-patch review is recommended.

### CODE REVIEW

**Comment by thomaseizinger** [comment]
I hope (:crossed_fingers:) that the search-replace changes I've made to the paths for Swift work like this. @roop Am I correct thinking that if CI builds then we are all gucci?

**Review by roop — APPROVED** [review]
@thomaseizinger Yes, given that CI passes, we're good to go. 👍🏻

File: swift/apple/FirezoneNetworkExtension/Adapter.swift [hunk 1]
```diff
@@ -92,10 +92,10 @@ public class Adapter {
   // Docs on filter strings: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.EnvFilter.html
   #if DEBUG
     private let logFilterString =
-      "connlib_apple=debug,firezone_tunnel=trace,libs_common=debug,firezone_client_connlib=debug,warn"
+      "connlib_client_apple=debug,firezone_tunnel=trace,connlib_shared=debug,connlib_client_shared=debug,warn"
   #else
     private let logFilterString =
-      "connlib_apple=info,firezone_tunnel=info,libs_common=info,firezone_client_connlib=info,warn"
+      "connlib_client_apple=info,firezone_tunnel=info,connlib_shared=info,connlib_client_shared=info,warn"
   #endif
```
Comment (roop) [inline swift/apple/FirezoneNetworkExtension/Adapter.swift:99]: @thomaseizinger Are the logs getting created correctly?

**Review by pratikvelani — APPROVED** [review]

**Review by conectado — COMMENTED** [review]
The changes looks good but I don't see the CI running in this branch

**Review by jamilbk — APPROVED** [review]
LGTM, but might need some tweaking after #2233 is merged.

**Comment by thomaseizinger** [comment]
> The changes looks good but I don't see the CI running in this branch

It was running yesterday!

**Comment by thomaseizinger** [comment]
> The changes looks good but I don't see the CI running in this branch

I think something got borked on GitHub's end because a link to a CI run I still had open in my browser returned a 404 so they might have lost some state?

Anyway, I amended the last commit and force-pushed to re-trigger it. It is running now :)

**Comment by thomaseizinger** [comment]
Merging because CI is green and @conectado agrees with the changes. @AndrewDryga is only tagged as reviewer because I am touching the workflow files. Given that CI runs, those changes should be good.

**Pull request merged**